### DISCUSSION TITLE
How to create migrations for permissions changes?

### DISCUSSION THREAD

**Discussion by eduardBM** [hasura/graphql-engine#8203] [Question] [answered]
Hi,
We're using hasura console migrate api to track migrations.
Recently we added a script to generate permissions changes to several tables dynamically, using "create_select_permission" json.

Running this script through the migrate api doesn't apply the permissions nor does it create the permissions.

Is there a way to have the permission changes create migrations using the migrations api?

**Comment by BenoitRanque** [comment, marked as answer]
The best way to achieve this would be to generate/change the metadata files, then apply those files using your standard CI/CD process.
This project [here](https://github.com/BenoitRanque/hasura-metadata-merge#additional-utilities) includes a utility to write/read the metadata into a JS object, you could use it to help you write your permissions.

**Reply by eduardBM** [reply]
Hi, thanks for you reply.

We were able to figure out what was wrong and now we get the correct metadata changes (i incorrectly assumed it would generate migration files, but it changes tables.yaml in-place).
